apps/src/templates/progress/progressHelpers.js

```diff
@@ -267,9 +267,10 @@ export function lessonProgressForSection(sectionLevelProgress, lessons) {
  * script.lessons.levels) contains more data than we need. This parses the parts
  * we care about to conform to our `levelType` object.
  */
-export const processedLevel = level => {
+export const processedLevel = (level, parentLevelId) => {
+  const id = level.activeId || level.id;
```

Comment:
Yeah this was confusing for me too :) not 100% sure what the multiple IDs refer to, but from I've seen, these levels typically just have one id in the `ids` field. Worth calling out that the "level" model passed into this function typically represents a script level, and this function converts it into a type that matches the `Level` interface in progressTypes. It's also a bit confusing because types are currently split between [progressTypes.ts](https://github.com/code-dot-org/code-dot-org/blob/5f3acce40a0b26755c9bd762443ea391e2aaa280/apps/src/types/progressTypes.ts#L4) and [progressTypes.js](https://github.com/code-dot-org/code-dot-org/blob/5f3acce40a0b26755c9bd762443ea391e2aaa280/apps/src/templates/progress/progressTypes.js#L4) so I think we need to do some cleanup to reconcile these with the actual shape used in these files.

Reply:
Did a little digging, and see that `activeId` seems to be referring to the active `script_level` variant.  https://codedotorg.atlassian.net/browse/PLAT-371  includes plan to add new active variant to a script level. Interesting [Slack thread](https://codedotorg.slack.com/archives/C02EEGWLHR8/p1641330249083000?thread_ts=1641321218.080800&cid=C02EEGWLHR8) in which @davidsbailey discusses how we want to get rid of level variants, but 'need other ideas on how to support fixing a badly broken level in a unit that already has progress in it.'